lib/internal/Magento/Framework/Backup/Db.php

```diff
@@ -38,14 +42,16 @@ public function rollback()
 
         $this->_lastOperationSucceed = false;
 
-        $archiveManager = new \Magento\Framework\Archive();
+        $archiveManager = new Archive();
         $source = $archiveManager->unpack($this->getBackupPath(), $this->getBackupsDir());
 
-        $file = new \Magento\Framework\Backup\Filesystem\Iterator\File($source);
+        $file = new File($source);
         foreach ($file as $statement) {
             $this->getResourceModel()->runCommand($statement);
         }
-        @unlink($source);
+        if ($this->keepSourceFile()) {
```

Comment:
The logic seems to be reversed here and should be !$this->keepSourceFile()

Reply:
I will check tomorrow, maybe I was wrong copying between branches

Reply:
Thanks!